src/main/java/net/coderbot/iris/mixin/shadows/MixinPreventRebuildNearInShadowPass.java

```diff
@@ -28,4 +36,14 @@
 			callback.cancel();
 		}
 	}
+
+	@Inject(method = "setupTerrain", at = @At("TAIL"))
+	private void iris$copyBlockEntities(Camera camera, Frustum frustum, boolean hasForcedFrustum, int frame, boolean spectator, CallbackInfo ci) {
+		ShadowRenderer.visibleBlockEntities.clear();
+		for (WorldRenderer.ChunkInfo chunk : this.visibleChunks) {
+			for (BlockEntity entity : ((ChunkInfoAccessor) chunk).getChunk().getData().getBlockEntities()) {
+				ShadowRenderer.visibleBlockEntities.add(entity);
+			}
+		}
```

Comment:
This should check whether the shadow pass is active before computing visibility info.

Reply:
Fixed